SLHCUpgradeSimulations/Phase2TrackerDTCCablingMap/plugins/DTCCablingMapTestProducer.cc

```diff
@@ -0,0 +1,113 @@
+// -*- C++ -*-
+//
+// Package:    SLHCUpgradeSimulations/Phase2TrackerDTC
+// Class:      DTCCablingMapTestProducer
+// 
+/**\class DTCCablingMapTestProducer DTCCablingMapTestProducer.cc SLHCUpgradeSimulations/Phase2TrackerDTC/plugins/DTCCablingMapTestProducer.cc
+
+Description: [one line class summary]
+
+Implementation:
+		[Notes on implementation]
+*/
+//
+// Original Author:  Luigi Calligaris, SPRACE, São Paulo, BR
+// Created        :  Wed, 27 Feb 2019 21:41:13 GMT
+//
+//
+
+#include <memory>
+#include <unordered_map>
+#include <utility>
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/one/EDAnalyzer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+
+
+#include "CondCore/DBOutputService/interface/PoolDBOutputService.h"
+#include "CondFormats/Common/interface/Time.h"
+
+#include "CondFormats/Phase2TrackerObjects/interface/TrackerDetToDTCELinkCablingMap.h"
+#include "CondFormats/Phase2TrackerObjects/interface/DTCELinkId.h"
+#include "CondFormats/DataRecord/interface/TrackerDetToDTCELinkCablingMapRcd.h"
+
+
+class DTCCablingMapTestProducer : public edm::one::EDAnalyzer<>
+{
+	public:
+		explicit DTCCablingMapTestProducer(const edm::ParameterSet&);
+		~DTCCablingMapTestProducer();
+		
+		static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+		
+		
+	private:
+		virtual void beginJob() override;
+		virtual void analyze(const edm::Event&, const edm::EventSetup&) override;
+		virtual void endJob() override;
+private:
+		cond::Time_t iovBeginTime_;
+		std::string  recordName_;
+};
+
+void DTCCablingMapTestProducer::fillDescriptions(edm::ConfigurationDescriptions& descriptions)
+{
+	edm::ParameterSetDescription desc;
+	desc.setComment("Stores a dummy TrackerDetToDTCELinkCablingMap database object from a CSV file.");
+  desc.add<long long unsigned int>("iovBeginTime", 1);
+  desc.add<std::string>("record","TrackerDetToDTCELinkCablingMap");
+	descriptions.addDefault(desc);
+}
+
+DTCCablingMapTestProducer::DTCCablingMapTestProducer(const edm::ParameterSet& iConfig):
+	iovBeginTime_(iConfig.getParameter<long long unsigned int>("iovBeginTime")),
+	recordName_  (iConfig.getParameter<std::string>("record"))
+{
+	
+}
+
+void DTCCablingMapTestProducer::beginJob()
+{
+	using namespace edm;
+	using namespace std;
+	
+	TrackerDetToDTCELinkCablingMap* pCablingMap = new TrackerDetToDTCELinkCablingMap();
```

Comment:
where does this pointer get deleted?

Reply:
Well, spotted. I forgot to delete.

Reply:
And the same lack of delete is in the actual producer, apart from the test one.

Reply:
In general, please use smart pointers (e.g. `unique_ptr`). There is almost no reason to use `new` in modern C++.

Reply:
Shouldn't you use `new` anyway in the smart pointer constructor, to pass the pure pointer to it?

Reply:
No, use `make_unique`

Reply:
The issue here is not the use of new in itself, but the fact that you use a raw pointer.
Off-topic: It is not recommended to use new in the smart pointer constructor either anyway. Better use std::make_unique (from C++14) and std::make_shared (already there in C++11).

Reply:
I ignored the existence of make_unique, thanks for the tip!